airbyte-integrations/connectors/source-teradata/src/test/java/io/airbyte/integrations/source/teradata/TeradataTestDatabase.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.teradata;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.cdk.db.factory.DatabaseDriver;
+import io.airbyte.cdk.db.jdbc.JdbcUtils;
+import io.airbyte.cdk.testutils.NonContainer;
+import io.airbyte.cdk.testutils.TestDatabase;
+import java.util.stream.Stream;
+import org.jooq.SQLDialect;
+
+public class TeradataTestDatabase extends TestDatabase<NonContainer, TeradataTestDatabase, TeradataTestDatabase.TeradataDbConfigBuilder> {
```

Comment:
Consider defining a `NonContainerTestDatabase` class in the CDK which extends `TestDatabase<NonContainer, NonContainerTestDatabase, NonContainerTestDatabase.NonContainerTestDatabaseConfigBuilder>` and use that here and in other connectors which don't rely on testcontainers. Inject the username, password, sql dialect etc. as constructor args.